### PR TITLE
update-vendor-firmware: various fixes to support firmware installation on boot in Fedora

### PR DESCRIPTION
**Pull request by davide125**
While testing https://github.com/AsahiLinux/asahi-scripts/pull/9 I've noticed `tar` fails with a very non obvious "file exists" error if the filesystem is read only. Passing `--overwrite` makes is fail explicitly with a read only fs error, which is much more useful. I've checked and busybox tar supports these options just fine.

### CODE REVIEW

**Review by Conan-Kudo — APPROVED** [review]

File: update-vendor-firmware [hunk 1]
```diff
@@ -57,7 +57,7 @@ if [ -e "$TARGET_MANIFEST" ]; then
 fi
 
 echo "Extracting updated vendor firmware..."
-tar xf "$VENDORFW/firmware.tar"
+tar -xf "$VENDORFW/firmware.tar" --overwrite
```
Comment (Conan-Kudo) [inline update-vendor-firmware:60]: Not every variant of `tar` will handle options at the end of the command. You should put `--overwrite` before the positional argument of the tarball filepath.

**Review by Conan-Kudo — CHANGES_REQUESTED** [review]

File: update-vendor-firmware (hunk 1 above)
Comment (davide125) [reply update-vendor-firmware]: Fixed, thanks

**Review by ericcurtin — APPROVED** [review]

**Comment by davide125** [comment]
Added a couple more commits based on the discussion in #9

**Review by ericcurtin — APPROVED** [review]

File: update-vendor-firmware (hunk 1 above)
Comment (ericcurtin) [reply update-vendor-firmware]: Wanna mark this as approved @Conan-Kudo ? The --overwrite was moved :)

**Pull request closed without merging**